client/extensions/woocommerce/app/order/order-details.js

```diff
@@ -24,13 +27,57 @@ class OrderDetails extends Component {
 	}
 
 	state = {
-		showRefundDialog: false,
+		status: this.props.order.status || false,
```

Comment:
I think once initial state is set based on props, this needs to move to a `constructor`.